nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/audit/ProcessorAuditor.java

```diff
@@ -400,4 +444,39 @@ private PropertyDescriptor locatePropertyDescriptor(Set<PropertyDescriptor> prop
         return specDescriptor;
     }
 
+    /**
+     * Gets Item Paths and set path and node in Map map
+     * @param nl NodeList to generate path
+     * @param path String path to ParentNode
+     * @param map  Map of path to node, and node reference
+     */
+    private void getItemPaths(NodeList nl, String path, Map<String,Node> map){
+        if(nl!=null) {
+
+
+            for (int i = 0; i < nl.getLength(); i++) {
+                Node n;
+                if (( n = nl.item(i)) != null) {
+                    if(n.getNodeType() == Node.ELEMENT_NODE || n.getNodeType() == Node.TEXT_NODE) {
+                        if(n.hasChildNodes()){
+                           if(n.getNodeType() == Node.ELEMENT_NODE) {
+                                getItemPaths(n.getChildNodes(), path + ":" + n.getNodeName(), map);
+                           }
+                        }
+                        if(!n.hasChildNodes()) {
+                            map.put(path + ":" + n.getNodeName().trim()+":"+n.getNodeValue(), n);
+                        }
+
+                        if (n.hasAttributes()) {
+                            NamedNodeMap na = n.getAttributes();
+                            for (int j = 0; j < na.getLength(); j++) {
+                                map.put(path + ":" + n.getNodeName() + ":" + na.item(j).getNodeName().trim()+":"+na.item(j).getNodeValue(), n);
```

Comment:
I'm seeing something like this displayed:
`:#document:criteria:rules:conditions:expression:#text:${filename:equals("hello!")}`
Is that intended? I would think it would be more helpful (if possible) to have something like the Rule Name then the condition or action or whatever changed.

Reply:
Short answer is yes. The issue being that annotation structure varies between processor types so presentation schema would have to be carried through to view the changes in context. I do not believe the model currently supports. This approach, while not near perfect,  allows viewing breadcrumbs that indicate changes negatively impacting flow.